benchmarks/nel/scripts/datasets/reddit.py

```diff
@@ -0,0 +1,157 @@
+
+    def _create_annotated_docs(self) -> List[Doc]:
+        annotated_docs: List[Doc] = []
+        file_names: List[str] = []
+        if self._options["posts"]:
+            file_names.append("posts.tsv")
+        if self._options["comments"]:
+            file_names.append("comments.tsv")
+        assert file_names, "Either 'posts' or 'comments' have to be True in corpus config."
+
+        # Join records with line breaks.
+        rows: List[List[str]] = []
+        for file_name in [self._paths["assets"] / file_name for file_name in file_names]:
+            row_length = 3 if file_name.name.endswith("posts.tsv") else 5
+            with open(file_name, encoding="utf-8") as file_path:
+                for row in csv.reader(file_path, delimiter="\t"):
+                    assert len(row) <= row_length
+                    # If row has fewer than the specified number of entries: newlines from comments have been
+                    # maintained, content is part of last valid comment.
+                    if 0 < len(row) < row_length:
+                        assert len(row) <= 1
+                        rows[-1][-1] += " " + row[0]
+                    elif len(row) == row_length:
+                        rows.append(row)
+
+        # Create spans from annotations.
+        for row in rows:
+            doc = self._nlp_base(row[-1])
+            # There might be multiple annotations for the same tokens/spans. This is handled by (1) sorting all
+            # entities for this document by their frequency and (2) afterwards moving all overlapping entities to
+            # the doc's _ attribute, so we might still consider that during evaluation.
+            # Additionally, there is a number of index errors in the annotations (especially in the bronze dataset).
+            # Some of these are resolved by aligning annotation with token indices.
+            doc.ents, overlapping_doc_annotations = _create_spans_from_doc_annotation(
+                doc=doc,
+                entities_info=self._entities,
+                annotations=self._annotations.get(row[0], []),
+                entities_failed_lookups=self._failed_entity_lookups
+            )
+            doc._.overlapping_annotations = overlapping_doc_annotations
+            annotated_docs.append(doc)
+
+        return annotated_docs
+
+    def clean_assets(self) -> None:
+        to_remove = {
+            "bronze_comment_annotations.tsv": {}
+        }
+        to_replace = {
+            "bronze_comment_annotations.tsv": {
+                "`` How to Lose a Guy in 10 Days": "How to Lose a Guy in 10 Days",
+                "\"How to Lose a Guy in 10 Days": "\"How to Lose a Guy in 10 Days\"",
+                "Money\t325\t330\tmoney\n": "Money\t323\t328\tmoney\n"
+            }
+        }
+        encoding = "utf-8"
+
+        for file_name in to_replace:
+            lines: List[str] = []
+            with open(self._paths["assets"] / file_name, "r", encoding=encoding) as file:
```

Comment:
I think that utf-8 is the default and doesn't have to specified here in Python3?

Reply:
I stumbled over some encoding issues with some of the text in the Reddit files which were fixed by setting UTF-8 explicitly.

Reply:
Huh, I looked it up and UTF-8 is *not* the default - it uses the "preferred locale" set by the OS ([ref](https://docs.python.org/3.9/library/functions.html?highlight=open#:~:text=the%20default%20encoding%20is%20platform%20dependent). So that's fine.